pkg/storegateway/bucket_chunk_reader.go

```diff
@@ -65,7 +67,7 @@ func (r *bucketChunkReader) addLoad(id chunks.ChunkRef, seriesEntry, chunkEntry
 	}
 	r.toLoad[seq] = append(r.toLoad[seq], loadIdx{
 		offset:      off,
-		length:      length,
+		length:      util_math.Max(varint.MaxLen32, length), // If the length is 0, we need to at least fetch the length of the chunk.
```

Comment:
After adding tests, I discovered that the bucketChunkReader cannot handle 0 estimations even though 0 is a valid value according to this godoc

https://github.com/grafana/mimir/blob/14080d085561056f7ef8aa70a5641cb403491c3b/pkg/storegateway/series_refs.go#L184

This wasn't causing bugs because `length` never would have been 0 as it is always set to a non-zero value here
https://github.com/grafana/mimir/blob/14080d085561056f7ef8aa70a5641cb403491c3b/pkg/storegateway/series_refs.go#L973-L988